frontend/src/components/status-badge.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { type ReactNode } from "react";
+
+type TrackerColor = "red" | "green" | "yellow";
+
+interface StatusBadgeProps {
+  color: TrackerColor;
+  children: ReactNode;
+}
+
+export const StatusBadge: React.FC<StatusBadgeProps> = ({
+  color,
+  children
+}) => {
+  let borderColor = "";
+  let bgColor = "";
+  let textColor = "";
+  let roundedColor = "";
+
+  switch (color) {
+    case "red":
+      borderColor = "border-red-600";
+      bgColor = "bg-red-600 bg-opacity-10";
+      textColor = "text-statusred";
+      roundedColor = "bg-red-600";
+      break;
+    case "green":
+      borderColor = "border-green-600";
+      bgColor = "bg-green-600 bg-opacity-10";
+      textColor = "text-statusgreen";
+      roundedColor = "bg-green-600 ";
+      break;
+    case "yellow":
+      borderColor = "border-yellow-600";
+      bgColor = "bg-yellow-600 bg-opacity-10";
+      textColor = "text-statusyellow";
+      roundedColor = "bg-yellow-600";
+      break;
+    default:
+      break;
+  }
```

Comment:
use `cn` in `src/lib/utils` instead of this, the API of cn is simple :

```tsx
// this is an example, don't just blindly copy it
import { cn } from "~/lib/utils";

function StatusBadge() {
    return (
        <div className={cn("flex border md:w-fit ...", 
          {
             "border-yellow-600 text-statusgreen bg-green-600 border-green-600": color ===  "green",
             "border-red-600 bg-red-600 bg-opacity-10 text-statusred bg-red-600": color ===  "red",
              // etc...
           })}>
          ...
         </div>
     )
}
```